pkg/resourcecreator/istio.go

```diff
@@ -97,6 +90,30 @@ func ServiceRoleBinding(app *nais.Application) (*istio_crd.ServiceRoleBinding, e
 	}, nil
 }
 
+func ServiceRoleBindingPrometheus(app *nais.Application) ( serviceRoleBindingPrometheus *istio_crd.ServiceRoleBinding, err error) {
+	if !app.Spec.Prometheus.Enabled {
+		return nil, nil
+	}
+
+	serviceRoleBindingPrometheus, err = ServiceRoleBinding(app)
```

Comment:
I don't know if I like this transmutation of an object that has a different responsibility. Maybe then it would be better to have a base function that is specifically designed for this purpose. This looks like potential buggy behavior in the future.

Reply:
Yeah I agree. But I thought this is what we agreed on to accomplish the task at hand. 

Do you want to redesign the functions in this issue or should we create a new one?

Reply:
See latest commit for clarification :)

Reply:
Fixed for ServiceRolePrometheus aswell :)